public/app/plugins/datasource/loki/modifyQuery.ts

```diff
@@ -540,14 +540,14 @@ function addLabelFormat(
   return newQuery;
 }
 
-export function addLineFilter(query: string): string {
+export function addLineFilter(query: string, value = ''): string {
   const streamSelectorPositions = getStreamSelectorPositions(query);
   if (!streamSelectorPositions.length) {
     return query;
   }
   const streamSelectorEnd = streamSelectorPositions[0].to;
 
-  const newQueryExpr = query.slice(0, streamSelectorEnd) + ' |= ``' + query.slice(streamSelectorEnd);
+  const newQueryExpr = query.slice(0, streamSelectorEnd) + ` |= \`${value}\`` + query.slice(streamSelectorEnd);
```

Comment:
I think this needs some escaping of the value:

https://github.com/grafana/grafana/assets/8092184/84fe22a6-e6ec-4cb5-8c84-f57684fc4c9c

Reply:
Good catch! Will do in the other PR.